src/Config/RuntimeConfigLoader.cs

```diff
@@ -96,6 +105,65 @@ public static bool TryParseConfig(string json, [NotNullWhen(true)] out RuntimeCo
         return true;
     }
 
+    /// <summary>
+    /// Parses a JSON string into a <c>RuntimeConfig</c> object for multiple database scenario.
+    /// </summary>
+    /// <param name="json">JSON that represents the config file.Json expected to contain Datasources dictionary.</param>
+    /// <param name="config">The parsed config, or null if it parsed unsuccessfully.</param>
+    /// <returns>True if the config was parsed, otherwise false.</returns>
+    public static bool TryParseConfigMultipleDatabase(string json, [NotNullWhen(true)] out RuntimeConfig? config, ILogger? logger = null, Dictionary<string, string>? datasourceNameToConnectionString = null)
+    {
+        JsonSerializerOptions options = GetSerializationOptions();
+
+        try
+        {
+            config = JsonSerializer.Deserialize<RuntimeConfig>(json, options);
+
+            if (config is null)
+            {
+                return false;
+            }
+
+            if (datasourceNameToConnectionString != null)
+            {
+                foreach ((string dataSourceName, string connectionString) in datasourceNameToConnectionString)
```

Comment:
When its a single database scenario, why cant the same function still be useful? This loop just run once, right? Couldn't you pass in a dictionary of the single datasource name to connection string as an argument?

Reply:
You could, ive just created a separate method to maintain backward compatibility.

Reply:
fixed after offline discussion.